pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -86,7 +87,12 @@ func (u UsernamePassword) Setup(baseDir string) error {
 	fmt.Fprintf(gitconfig, UserPassGitConfig, gitcredentials.Name())
 	fmt.Fprintf(gitcredentials, "%s", usernamePasswordURL.String())
 
-	return ensureGitConfigIncludes(gitconfig.Name())
+	sourceURL := u.SourceURL
+	sourceURL.User = nil
```

Comment:
username doesn't have to match from the URL? I couldn't tell from the documentation